src/meta/src/hummock/compactor_manager.rs

```diff
@@ -157,23 +206,117 @@ impl CompactorManager {
         max_concurrent_task_number: u64,
     ) -> Receiver<MetaResult<SubscribeCompactTasksResponse>> {
         let (tx, rx) = tokio::sync::mpsc::channel(STREAM_BUFFER_SIZE);
+        self.remove_compactor(context_id);
         let mut guard = self.inner.write();
-        guard.compactors.retain(|c| c.context_id != context_id);
-        guard.compactors.push(Arc::new(Compactor {
+        guard.compactors.push(context_id);
```

Comment:
It's possible two concurrent `add_compactor` both `self.remove_compactor` and then `guard.compactors.push`, which causes duplicate items in `compactors`?

Reply:
Yes, perhaps I will make the operation atomic under a single guard to avoid concurrency issues.